src/components/Composer/index.js

```diff
@@ -122,380 +123,317 @@ const IMAGE_EXTENSIONS = {
+            }}
+        >
+            <Text
+                multiline
+                style={[StyleSheet.flatten([style, styles.noSelect]), numberOfLines < props.maxLines ? styles.overflowHidden : {}, {maxWidth: textInputWidth}]}
             >
+                {`${valueBeforeCaret} `}
                 <Text
-                    multiline
-                    style={[propStyles, this.state.numberOfLines < this.props.maxLines ? styles.overflowHidden : {}, {maxWidth: this.state.width}]}
+                    numberOfLines={1}
+                    ref={textRef}
                 >
-                    {`${this.state.valueBeforeCaret} `}
-                    <Text
-                        numberOfLines={1}
-                        ref={this.textRef}
-                    >
-                        {`${this.state.caretContent}`}
-                    </Text>
+                    {`${caretContent}`}
                 </Text>
-            </View>
-        );
-
-        // We're disabling autoCorrect for iOS Safari until Safari fixes this issue. See https://github.com/Expensify/App/issues/8592
-        return (
-            <>
-                <RNTextInput
-                    autoComplete="off"
-                    autoCorrect={!Browser.isMobileSafari()}
-                    placeholderTextColor={themeColors.placeholderText}
-                    ref={(el) => (this.textInput = el)}
-                    selection={this.state.selection}
-                    onChange={this.shouldCallUpdateNumberOfLines}
-                    style={[
-                        propStyles,
-
-                        // We are hiding the scrollbar to prevent it from reducing the text input width,
-                        // so we can get the correct scroll height while calculating the number of lines.
-                        this.state.numberOfLines < this.props.maxLines ? styles.overflowHidden : {},
-                        StyleUtils.getComposeTextAreaPadding(this.props.numberOfLines),
-                    ]}
-                    /* eslint-disable-next-line react/jsx-props-no-spreading */
-                    {...propsWithoutStyles}
-                    onSelectionChange={this.addCursorPositionToSelectionChange}
-                    numberOfLines={this.state.numberOfLines}
-                    disabled={this.props.isDisabled}
-                    onKeyPress={this.handleKeyPress}
-                />
-                {this.props.shouldCalculateCaretPosition && renderElementForCaretPosition}
-            </>
-        );
-    }
+            </Text>
+        </View>
+    );
+
+    return (
+        <>
+            <RNTextInput
+                autoComplete="off"
+                autoCorrect={!Browser.isMobileSafari()}
+                placeholderTextColor={themeColors.placeholderText}
+                ref={setTextInputRef}
+                selection={selection}
+                style={[
+                    StyleSheet.flatten([style, {outline: 'none'}]),
```

Comment:
In some places, implementation is slightly different. Could you explain why these changes? There is no outline: 'none', old code.

Reply:
We use the outline in the old code [here](https://github.com/Expensify/App/pull/23359/files#diff-224d9ca075668f353172108020027fb452485426ce8263e0070e71442a642297L442)
do you mean using useMemo for this?

Reply:
Thanks, it's ok, I couldn't find this style because git diff changes are going crazy, harder to compare

Reply:
Yeah, sometimes it just gets too messy 😄